### ISSUE TITLE
Invoke-Reboot and Server Core

### ISSUE THREAD

**Issue by LukeCarrier** [mwrock/boxstarter#215]
First of all, thanks for the work you're doing with this module -- it's an absolute lifesaver.

I've [recently been attempting to package Server Core](https://github.com/LukeCarrier/vagrant-windows/tree/editions) and appear to have hit a roadblock with the `Invoke-Reboot` cmdlet. It seems that Core editions don't execute items in the Startup folder, so `boxstarter-post-restart.bat` never runs.

I've managed to work around this by setting `PowerShell.exe -NoExit` as the `Winlogon\Shell` value and adding the following to my `$PROFILE` prior to invoking a reboot:

```powershell
try {
    $startup = "$($env:APPDATA)\Microsoft\Windows\Start Menu\Programs\Startup\boxstarter-post-restart.bat"
    if (!$env:PACKER_STARTUP_WORKAROUND -and (Test-Path $startup)) {
        & cmd /c $startup
    }
} finally {
    $env:PACKER_STARTUP_WORKAROUND = 1
}
```

I realise this is _really_ ugly. Any ideas how to do it "properly"?

**Comment by mwrock** [comment]
Yeah unfortunately the way boxstarter handles startup after reboot is not core friendly at all. Ideally I think boxstarter should use a scheduled task to start back up instead of the startup folder.

**Comment by MarkDordoy** [comment]
Thanks for opening this Issue @LukeCarrier. I'm fairly new to packer but got it working with server 2016. I'm now looking at the core version and run into the same problems it looks like you ran into. 

I am grateful for your help regarding the invoke-reboot issue and your workaround. 👍 🥇 

@mwrock do you plan to support core with boxstarter regarding invoke-reboot

**Comment by aaronk1** [comment]
Has there been any update on this one, @mwrock ?

**Comment by mwrock** [comment]
No updates here. Have not had spare time to devote to this but happy to review PRs if they come in.

**Comment by username-is-already-taken2** [comment]
Hello @mwrock , I can't promise anything but I might be able to take a punt on this.  
Before I code myself to far down a raddit hole, could I get a steer from you?

I've used scheduled tasks on core to automatically kickoff a script after a reboot using the onlogon trigger and flagging it for interaction with the desktop. that works pretty well but I do need to know the credentials of the user automatically logging in when creating the scheduled task.

_Questions_

* Could I use the approach linked below to get the automatic user credientials?

https://github.com/mwrock/boxstarter/blob/90c53cfdbe0e78b55d2d27e724375afffa27894d/Boxstarter.Bootstrapper/Cleanup-Boxstarter.ps1#L23

* Where do you see the logic of creating the schededtask going into? I was thinking this function `boxstarter/Boxstarter.Bootstrapper/Invoke-Reboot.ps1` ?

Appreciate your help

Best Regards

Gary

**Comment by mwrock** [comment]
Boxstarter.autologon stores a backup of any previous auto logon registry values and the referenced code restores those values during the boxstarter cleanup phase. You would want to create the task around the same time it calls `Set-SecureAutoLogon` around the end of `Cleanup-Boxstarter.ps1`. You would want to create one task that is potentially reused accross multiple reboots. After the first reboot, `$BoxstarterPassword` is no longer available.

**Comment by kshef** [comment]
Very interested in the resolution to this issue. Is autologon required for boxstarter to resume after a reboot? Or could the scheduled task (with stored credentials) resume execution without an interactive session? In my use case, autologon might not be possible due to group policy.

**Comment by mwrock** [comment]
In theory I think it could pick it up in the task non-interactively. The question is how would a user ever know boxstarter is done and also not lose visibility into the success failure without polling the log file? The approach would have to consider that bit of UX.

**Comment by LukeCarrier** [comment]
This issue is a decade old, and I don't intend to address it as I no longer operate Windows Server. Closing.
